### PR TITLE
Feature #16 - Share identity DOM tests

### PR DESCRIPTION
**Pull request by albertandrejev**
**Description**
This PR will add DOM tests and route for the Share Identity view. It will implement "Add route and implement DOM tests" part of the #16.

### CODE REVIEW

File: packages/medulas-react-components/src/components/forms/CheckboxField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const onCheckBoxChange = (_: React.ChangeEvent<HTMLInputElement>, checked: boolean): void => {
+    onChange(checked);
+    if (onChangeCallback) {
+      onChangeCallback(checked);
+    }
+  };
+
+  const control = (
+    <Checkbox
+      checked={value === '' ? false : value}
```
Comment (apanizo) [inline packages/medulas-react-components/src/components/forms/CheckboxField/index.tsx]: What is value? a string? Maybe you just want to do `checked={!!value}` ?

File: packages/sanes-chrome-extension/src/routes/share-identity/test/operateShareIdentity.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+export const clickOnAcceptButton = async (ShareIdentityDom: React.Component): Promise<void> => {
+  const inputs = TestUtils.scryRenderedDOMComponentsWithTag(ShareIdentityDom, 'button');
+
+  expect(inputs.length).toBe(2);
+
+  const acceptButton = inputs[0];
+
+  TestUtils.act(
+    (): void => {
```
Comment (webmaster128) [inline packages/sanes-chrome-extension/src/routes/share-identity/test/operateShareIdentity.ts]: We changed the linter settings, such that the return type only needs to be set when the function has a name. Especially in test code, you don't need to set it. This saves you 2 lines and one indent level:

```ts
TestUtils.act(() => {
  TestUtils.Simulate.click(acceptButton);
});
```

File: packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  let identityDOM: React.Component;
+
+  beforeEach(async () => {
+    store = aNewStore();
+    identityDOM = await travelToShareIdentity(store);
+  });
+
+  it('should accept incoming request and come back', async (): Promise<void> => {
+    await clickOnAcceptButton(identityDOM);
+    await confirmSelectionButton(identityDOM);
```
Comment (apanizo) [inline packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts]: Let's follow the conversation in the front-end channel, but I do believe this screen is not necessary.

File: packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    //TODO: Check here that share request rejection has been reject successfuly
+
+    /**
+     * Remove this code if not required in case if there is another redirection
+     * in confirmAcceptButton method. And apply this method in separate test method.
+     */
+    await clickOnBackButton(identityDOM);
```
Comment (apanizo) [inline packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts]: What are you trying to do here? as far as I see in the invision mock, the final redirection is to the account status view. 🤔

Comment (albertandrejev) [reply packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts]: Ok, will check for account status view

Comment (albertandrejev) [reply packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts]: Hmmm.. where to redirect in case of requests queue?

Comment (albertandrejev) [reply packages/sanes-chrome-extension/src/routes/share-identity/index.dom.spec.ts]: I got the answer

> Once the request is dispatched, the view should be redirected to: `request view` if there are more pending requests, or to `account-status view` if nothing is left.

**Pull request merged**